packages/web-api-scan-runner/src/report-generator/axe-result-to-consolidated-html-converter.ts

```diff
@@ -1,38 +1,39 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-
-import { AxeReportParameters, ReporterFactory } from 'accessibility-insights-report';
-import { AxeResults } from 'axe-core';
+import { ReporterFactory } from 'accessibility-insights-report';
+import axe from 'axe-core';
 import { inject, injectable } from 'inversify';
-import { ReportFormat } from 'storage-documents';
+import { ReportFormat, CombinedScanResults } from 'storage-documents';
+import { CombinedReportDataConverter, ScanResultData } from 'axe-result-converter';
 import { iocTypeNames } from '../ioc-types';
-import { AxeResultConverter, ReportGenerationParams } from './axe-result-converter';
 import { htmlReportStrings } from './html-report-strings';
+import { AxeResultConverterOptions } from './axe-result-converter';
 
 @injectable()
-export class AxeResultToConsolidatedHtmlConverter implements AxeResultConverter {
+export class AxeResultToConsolidatedHtmlConverter {
     public readonly targetReportFormat: ReportFormat = 'consolidated.html';
 
-    constructor(@inject(iocTypeNames.ReporterFactory) private readonly reporterFactoryFunc: ReporterFactory) {}
+    constructor(
+        @inject(CombinedReportDataConverter) private readonly combinedReportDataConverter: CombinedReportDataConverter,
+        @inject(iocTypeNames.ReporterFactory) private readonly reporterFactoryFunc: ReporterFactory,
+        private readonly axeCore: typeof axe = axe,
+    ) {}
 
-    public convert(results: AxeResults, params: ReportGenerationParams): string {
+    public convert(combinedScanResults: CombinedScanResults, options: AxeResultConverterOptions): string {
         const reporter = this.reporterFactoryFunc();
-
-        const htmlReportParams: AxeReportParameters = {
-            results: results,
-            description: this.createDescription(results),
-            serviceName: htmlReportStrings.serviceName,
-            scanContext: {
-                pageTitle: params.pageTitle,
-            },
+        const scanResultData: ScanResultData = {
+            baseUrl: options.baseUrl ?? 'n/a',
+            basePageTitle: '', // not used
```

Comment:
page title is used by the report package for the tooltip/accessible name of the base page link. If we're using this data directly for report generation, can we add the page title here, or if this is an intermediate data type can we remove basePageTitle from ScanResultData?

Reply:
not deterministic value at this moment, we do not scan baseUrl hence we do not have corresponding page title. We can have either empty string or fixed text here. We cannot remove property b/c it is used by CLI package.